lib/src/ui/pages/location/location_notifier.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:allthenews/src/domain/location/get_location_use_case.dart';
+import 'package:allthenews/src/domain/location/location.dart';
+import 'package:allthenews/src/domain/location/location_exception.dart';
+import 'package:allthenews/src/domain/location/location_resolver.dart';
+import 'package:allthenews/src/ui/pages/location/location_error_view_entity_mapper.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:geolocator/geolocator.dart';
+
+class LocationNotifier extends ChangeNotifier {
+  final GetLocationUseCase _getLocationUseCase;
+  final LocationResolver _locationResolver;
+  final LocationErrorViewEntityMapper _errorViewEntityMapper;
+
+  LocationViewState _state = const LocationViewState();
+
+  LocationNotifier(this._getLocationUseCase, this._errorViewEntityMapper, this._locationResolver);
+
+  LocationViewState get state => _state;
+
+  Future<void> findUserLocation() async {
+    _setNotifierState(const LocationViewState(isLoading: true));
+    await _getLocationUseCase()
+        .then((location) => _setNotifierState(LocationViewState(location: location)))
+        .catchError((exception) => _setNotifierState(
+            LocationViewState(error: _errorViewEntityMapper.map(exception as LocationException))));
+  }
```

Comment:
dobrze by było wybrać podejście do error handlingu, opcje są dwie:
- używanie API Futuresów i robienie `then` `catchError` i `whenComplete`,
- używanie try/catch

W tej metodzie wybrałeś oba podejścia i to async/await jest tu zbędnę. Ja byłbym jednak za tym, żeby używać podejścia z try/catch. Dlatego, że jest to nowsze i zalecane podejście do obsługi błędów, linki:
- https://dart.dev/guides/libraries/futures-error-handling
- https://dart.dev/guides/language/language-tour#asynchrony-support